site/app/libraries/Core.php

```diff
@@ -406,12 +406,18 @@ public function buildUrl($parts=array(), $hash = null) {
     }
 
     /**
+     * Given some URL parameters (parts), build a URL for the site using those parts
```

Comment:
I think the variable `$with_course_info` should be changed to something more descriptive, without reading the code I'm not sure what this actually does, I recommend something like `$prepend_course_info` or `$add_course_info`

I also think there should be a comment (outside of the @ param line) explicitly stating "this function will add the semester and course to the beginning of the new url by default".
I can see this being an area future developers accidentally misuse and run into getting a URL they're not expecting if the functions behaviour isn't well defined.

Reply:
`$prepend_course_info` sounds good to me. Sorry that as I am not a native speaker, I am not sure what is the best way to describe this. I do feel it is important to have sufficient comment on this function. Thank you very much for this advice. Will make the change.